micro: remove lite-specific code from copy of FILL

rkuester opened the pull request:
Remove the bulk of lite-specific code from the micro
implementation of operator FILL.
- Flatten the namespace
- Don't resize output tensors
- Remove input and output types other than int8 and float32
- Don't use gtest

This is PR4 in the sequence of PRs, outlined in issue #45306, to port operator FILL from lite to micro.

rkuester committed:
micro: remove lite-specific code from copy of FILL

Remove the bulk of lite-specific code from the micro
implementation of operator FILL.
- Flatten the namespace
- Don't resize output tensors
- Remove input and output types other than int8 and float32
- Don't use gtest

rkuester commented:
This PR is stacked atop PR #45457. It should be rebased and moved from draft to ready status as soon as that PR is merged.

petewarden commented:
There's a formatting error flagged in the TFLite Micro test:

```
Formatting errors found
--- /workspace/tensorflow/lite/micro/kernels/fill.cc  (original)
+++ /workspace/tensorflow/lite/micro/kernels/fill.cc  (reformatted)
@@ -85,8 +85,7 @@
       break;
     default:
       context->ReportError(
-          context,
-          "Fill only currently supports float32 for input 1, got %d.",
+          context, "Fill only currently supports float32 for input 1, got %d.",
           value->type);
       return kTfLiteError;
   }
```

It's also using context->ReportError, so we'll need to switch that to the macro instead too.

gbaned commented:
@rkuester This PR is in draft, can you please take a look on it. Thanks!

rkuester committed:
micro: fix formatting error in edit of op FILL

Fix a trivial formatting error in edit of op FILL that causes
pre-merge checks to fail.
